images/build/debian-iptables/distroless/get-binaries-from-package.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# USAGE: get-binaries-from-package.sh /opt/stage package1 package2
+# TODO: The above is not true. I will fix it!
+#
+# Stages all the packages and its dependencies (+ libraries and copyrights) to $1
+#
+# This is intended to be used in a multi-stage docker build with a distroless/base
+# or distroless/cc image.
+
+. package-utils.sh || exit 1 
+
+get_file_list() {
+    IFS="
+    "
+    REQUIRED_FILES="$(dpkg -L "${1}" | grep -vE '(/\.|/s?bin/|/usr/share/(man|doc|.*-completion))' | sed 's/\n/ /g')"
+    for file in $REQUIRED_FILES; do 
+        if [ -f "$file" ]; then
+            stage_file "${file}" "${STAGE_DIR}"
+        fi
+    done
+
+    BIN_LIST="$(dpkg -L "${1}" | grep -E '/s?bin/' |sed 's/\n/ /g')"
+    for binary in $BIN_LIST; do
+        /stage-binary-and-deps.sh "${binary}" "${2}"
```

Comment:
So this is a weird mix of things... now we copy everything except docs from the main package (`$1`), but for any shared libraries it depends on, we only copy the actual `.so` files, and not any other files from those packages...

Reply:
hum, actually what we do now is to check the dependencies of a package, and run the same "stage" function here. This stage function will copy everything from that package except man pages and binaries. The binaries and dynamic libraries will be copied later from that same binary.

In this case, I think this way: the majority of dependencies and files from those dependencies are part of the dpkg manifest. So you have iptables, and iptables depends on conntrack. You will make the full copy of iptables and conntrack packages.

One thing tho I didn't did here (and also because I was afraid of getting into a rabbit hole) is to check the dependency of the dependency. I expect that maybe when you get a dependency of a package, all the children dependencies are recovered as well.